cfme/common/provider.py

```diff
@@ -134,6 +134,16 @@ def create(self, cancel=False, validate_credentials=True, check_existing=False,
         else:
             created = True
 
+            # Enable alerts collection before adding the provider to avoid missing active
+            # alert after adding the provider
+            # For more info: https://bugzilla.redhat.com/show_bug.cgi?id=1514950
+
+            from cfme.containers.provider.openshift import OpenshiftProvider
+            if self.one_of(OpenshiftProvider) and \
```

Comment:
**Required**

Re-commenting since first post was hidden on `self.one_of()` update:


Beyond the use of provider.one_of, there is a more fundamental change required here @shalomnaim1.

This is very provider specific, and while below there is use of self.one_of(VmwareProvider) that is a bad example that should not be here either.

This should be in a new OpenshiftProvider.create() method, which makes this configuration change and then calls super(OpenshiftProvider, self).create() in order to call this method from the 'parent' class. No need to completely override the create method.

Happy to discuss on IRC/BlueJeans if its not clear how this would be done.